### ISSUE TITLE
Is there any plan to opensource the MPE related code in this repository?

### ISSUE THREAD

**Issue by Theohhhu** [uoe-agents/epymarl#5]
Awesome work for extending the original pymarl! But will be much better if MPE can be included.
Is there any plan to open source the MPE-related code in this repository?

**Comment by semitable** [comment]
Hi @Theohhhu 

Yes, this is a priority for us. Actually, the code is already public on my fork of MPE: https://github.com/semitable/multiagent-particle-envs

Essentially all this fork does (other than fixing some gym compatibility issues) is i) registering the environments with the gym interface when imported as a package and ii) correctly seeding the environments iii) makes the action space compatible with Gym (I think MPE originally does a weird one-hot encoding of the actions). 
There are also some extra environments but I would ignore them for now.

The fork's readme file hasn't been updated, but I think you can easily install it by cloning, and then running `pip install -e .` in the cloned folder (we include a `setup.py`).

The environment Gym names can be found in https://github.com/semitable/multiagent-particle-envs/blob/master/mpe/__init__.py#L6 and copied here:
```python
...
    "multi_speaker_listener": "MultiSpeakerListener-v0",
    "simple_adversary": "SimpleAdversary-v0",
    "simple_crypto": "SimpleCrypto-v0",
    "simple_push": "SimplePush-v0",
    "simple_reference": "SimpleReference-v0",
    "simple_speaker_listener": "SimpleSpeakerListener-v0",
    "simple_spread": "SimpleSpread-v0",
    "simple_tag": "SimpleTag-v0",
    "simple_world_comm": "SimpleWorldComm-v0",
...
```

Therefore, after installing that repository, one can use EPyMARL using something like:
```shell
python3 src/main.py --config=qmix --env-config=gymma with env_args.time_limit=50 env_args.key="mpe:SimpleSpeakerListener-v0"
```
When we test the installation instructions we can hopefully update the README to make it easier for everyone.
Also, if you encounter any issues with the above instructions, let us know here.

**Comment by semitable** [comment]
Oh, and EPyMARL sums the rewards of multiple agents to get a cooperative reward. The values reported in our paper are divided by the number of agents in the particular game.

**Comment by semitable** [comment]
We should close this issue only when we have updated the instructions for MPE.

**Comment by Theohhhu** [comment]
@semitable 
Awesome mate.  I've installed your fork and run the whole pipeline successfully. Thanks for your work and I think this repo will be a great benchmark for the MARL community in the future.

**Comment by to1a** [comment]
Thanks for your contribution, and could you provide some use cases on using pretrained adversary?

**Comment by semitable** [comment]
Hi! You are right, I forgot to mention those!
We define the pretrained ones in this folder: https://github.com/uoe-agents/epymarl/tree/main/src/pretrained (includes wrappers and trained parameters).

You can run them with something along the lines of:
```sh
python3 src/main.py --config=qmix --env-config=gymma with env_args.time_limit=25 env_args.key="mpe:SimpleAdversary-v0" env_args.pretrained_wrapper="PretrainedAdversary"
```
 for adversary and
```sh
python3 src/main.py --config=qmix --env-config=gymma with env_args.time_limit=25 env_args.key="mpe:SimpleTag-v0" env_args.pretrained_wrapper="PretrainedTag"
```
for tag.

**Comment by semitable** [comment]
Closes with 83a745bc9585b1f28dfe33e7752d19e511c7b385

**Comment by Gloriabhsfer** [comment]
Hi @semitable :

great jobs for the whole projects.  I have read the paper for this repositories and interested in the MPE simple-tag training parts. You mentioned you only focus the cooperative relationship in simple_tag and let the adversary groups' policy obtain form pretrained model. I have read the pretrained model in this projects. 

You have define your DDPG agent like this:

`class DDPG():
    def __init__(self, actor_input_dim, actor_output_dim, critic_input_dim, hidden, lr)`

In `Adversary.py ` the dimension is `DDPG(8,5,50,64,0.01)` and in `tag.py` is `DDPG(14,5,50,128,0,01)` I am not sure the dimension for the model that you provided in code as `adv_params.pt` and `prey_params.pt` related to which model? 
In your scenario setting, here are 3 adv and 1 good agents, I don't know why the `actor_output_dim` will be 5. Did you change any setting for your actionspace in this projects?

Best,
Gloria

**Comment by semitable** [comment]
Hi @Gloriabhsfer.
Did you install MPE from our fork (linked above?). If so, what command are you running?

**Comment by Gloriabhsfer** [comment]
> Hi @Gloriabhsfer. Did you install MPE from our fork (linked above?). If so, what command are you running?

Yes, I am running the command `python3 src/main.py --config=qmix --env-config=gymma with env_args.time_limit=25 env_args.key="mpe:SimpleTag-v0" env_args.pretrained_wrapper="PretrainedTag` which you provided in your tutorials.
